src/gloss/data/primitives.clj

```diff
@@ -71,6 +71,51 @@
    :be `ByteOrder/BIG_ENDIAN
    :ne `(ByteOrder/nativeOrder)})
 
+(defn bytes->umedium
+  [x bo]
+  (let [bo-trans (if (= :le bo) reverse identity)
+        [b0 b1 b2] (bo-trans x)]
+    (bit-or
+     (bit-or
+      (bit-shift-left (bit-and b0 0xFF) 16)
+      (bit-shift-left (bit-and b1 0xFF) 8))
+     (bit-shift-left (bit-and b2 0xFF) 0))))
+
+(defn umedium->medium
+  [x]
+  (let [uval2 (bit-and x 0x800000)]
+    (if (not= uval2 0)
+      (bit-or x 0xff000000)
+      x)))
+
+(defn medium->bytes
+  [x bo]
+  (let [bo-trans (if (= :le bo) reverse identity)]
+    (into-array Byte/TYPE
+                (bo-trans
+                 (map unchecked-byte
+                      [(bit-shift-right x 16)
+                       (bit-shift-right x 8)
+                       x])))))
+
+(defn get-unsigned-medium [buffer]
+  (let [bytes (byte-array 3)]
+    (.get ^ByteBuffer buffer bytes)
+    (bytes->umedium bytes :be)))
+
+(defn get-unsigned-medium-le [buffer]
+  (let [bytes (byte-array 3)]
+    (.get ^ByteBuffer buffer bytes)
+    (bytes->umedium bytes :le)))
+
+(defn put-medium [buffer value]
+  (let [bytes (medium->bytes value :be)]
+    (.put ^ByteBuffer buffer ^bytes bytes)))
+
+(defn put-medium-le [buffer value]
+  (let [bytes (medium->bytes value :le)]
+    (.put ^ByteBuffer buffer ^bytes bytes)))
+
```

Comment:
Hmm, won't the default versions fail if the ByteBuffer's ordering is LE? The built-in versions of things like `putInt` and `getLong` use the `.order()` of the ByteBuffer to interpret the next _X_ bytes, but `get-unsigned-medium` and `put-medium` assume they're big-endian, iiuc.

Reply:
I just refactored to use native byte order for default codec.